solr/core/src/java/org/apache/solr/core/snapshots/SolrSnapshotsTool.java

```diff
@@ -226,10 +226,9 @@ public void describeSnapshot(String collectionName, String snapshotName) {
       }
 
       String indexDirPath = coreSnap.getIndexDirPath();
-      if (pathPrefix.isPresent()) {
+      if (pathPrefix != null) {
         // If the path prefix is specified, rebuild the path to the index directory.
-        Path t = new Path(coreSnap.getIndexDirPath());
-        indexDirPath = (new Path(pathPrefix.get(), t.toUri().getPath())).toString();
+        indexDirPath = new Path(pathPrefix, coreSnap.getIndexDirPath()).toString();
```

Comment:
This is referring to Hadoop APIs! `org.apache.hadoop.fs.Path` Ugh!  Could it have been intentional and thus is there risk your change actually breaks something?

Reply:
Yea, this change is still fine, the old code is doing a bunch of extra conversions between Hadoop Path, URI, and String, and back and forth.